### PR TITLE
Implement Traffic Light detector

### PR DESCRIPTION
**Pull request by ksr1122**
* Implement main loop
* Implement KDTree for closest waypoint lookup

### CODE REVIEW

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -41,21 +47,30 @@ def __init__(self):
         self.upcoming_red_light_pub = rospy.Publisher('/traffic_waypoint', Int32, queue_size=1)
 
         self.bridge = CvBridge()
-        self.light_classifier = TLClassifier()
```
Comment (ashwinwadte) [inline ros/src/tl_detector/tl_detector.py:44]: please revert.

Comment (ksr1122) [reply ros/src/tl_detector/tl_detector.py:44]: Shall be reverted after the TrafficLight classifier is implemented.

Comment (ksr1122) [reply ros/src/tl_detector/tl_detector.py]: Reverted.

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -10,6 +10,9 @@
 import tf
 import cv2
 import yaml
+from scipy.spatial import KDTree
+from PIL import Image as im
```
Comment (ashwinwadte) [inline ros/src/tl_detector/tl_detector.py:14]: not needed.

Comment (ashwinwadte) [reply ros/src/tl_detector/tl_detector.py:14]: I am merging this change. We can get it removed in the next commit.

**Pull request merged**